op_builder/builder.py

```diff
@@ -261,7 +262,9 @@ def compute_capability_args(self, cross_compile_archs=None):
         args = []
         for cc in ccs:
             cc = cc.replace('.', '')
-            args.append(f'-gencode=arch=compute_{cc},code=compute_{cc}')
+            args.append(f'-gencode=arch=compute_{cc},code=sm_{cc}')
+            if cc.endswith('+PTX'):
+                args.append(f'-gencode=arch=compute_{cc},code=compute_{cc}')
```

Comment:
Shouldn't we use the compute-capability (cc) number here, rather than concatenated with '+PTX'?

Reply:
There are 2 unrelated things. 

1. The code you quoted above does the right thing if a user passes `TORCH_CUDA_ARCH_LIST` that contains `+PTX`, as my previous PR missed that part (I had no clue how +PTX was converted to compile flags at that time, but now I do)

2. The question is whether to add `+PTX` for jit_mode or not here:
https://github.com/microsoft/DeepSpeed/blob/03111cef09b33a25792246785ab0ded68be1733c/op_builder/builder.py#L246
I added it to sync with pytorch CUDAExtension implementation, which is how it'll do it once pytorch-1.8 is released. But you can choose to not include it. I have no idea how and when jit_mode is used.

So to conclude the code you quoted above does the right thing if any of the `ccs` contains '+PTX`, which may come from  user-defined `TORCH_CUDA_ARCH_LIST`- so it's a must. But wrt to (2) you can choose not to include it.
 If you do - please let me know and I will remove https://github.com/microsoft/DeepSpeed/blob/03111cef09b33a25792246785ab0ded68be1733c/op_builder/builder.py#L246

Let me know.

Reply:
oh, I see, there is a bug in my code - one second